jib-core/src/main/java/com/google/cloud/tools/jib/image/ImageLayers.java

```diff
@@ -19,34 +19,36 @@
 import com.google.common.collect.ImmutableList;
 import com.google.common.collect.ImmutableSet;
 import java.util.Iterator;
+import java.util.LinkedHashSet;
+import java.util.Set;
 import javax.annotation.Nullable;
 
 /** Holds the layers for an image. Makes sure that each layer is only added once. */
 public class ImageLayers<T extends Layer> implements Iterable<T> {
 
   public static class Builder<T extends Layer> {
 
-    private final ImmutableList.Builder<T> layersBuilder = ImmutableList.builder();
+    private final Set<T> layersBuilder = new LinkedHashSet<>();
```

Comment:
- Isn't the order of layers important? (`Set` does not provide any ordering guarantees.)
- Are you comfortable calling this a builder?

Reply:
I'm using `LinkedHashSet` to ensure the order is the order of insertion - I'll change the signature to `LinkedHashSet<T> layers` to make this clearer.